project/src/main/java/com/google/autograder/servlets/helpers/API.java

```diff
@@ -68,6 +71,27 @@ public static String getCurrentUserAPIAuthorization() {
         return authorization;
     }
 
+    public static HttpURLConnection getAuthenticatedRequest(String method, String endpoint) {
```

Comment:
Try to limit the use of `null` in your code, as it's very easy to forget to check for null. If you need to return something or nothing, use Optional (https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html), so it's clear that you're returning an `HttpURLConnection` or nothing and callers can deal it with appropriately

Reply:
Okay, interesting. I'll definitely look into this.